src/gui/SubWindow.cpp

```diff
@@ -26,42 +26,250 @@
+	QString clippedText = metrix.elidedText(text, Qt::ElideRight, width);
+	label->setText(clippedText);
 }
 
+
+
+
 QRect SubWindow::getTrueNormalGeometry() const
 {
 	return m_trackedNormalGeom;
 }
 
+
+
+
+QBrush SubWindow::activeColor() const
+{
+	return m_activeColor;
+}
+
+
+
+
+QColor SubWindow::textShadowColor() const
+{
+	return m_textShadowColor;
+}
+
+
+
+
+QColor SubWindow::borderColor() const
+{
+	return m_borderColor;
+}
+
+
+
+
+void SubWindow::setActiveColor(const QBrush & b)
+{
+	m_activeColor = b;
+}
+
+
+
+
+void SubWindow::setTextShadowColor(const QColor & c)
+{
+	m_textShadowColor = c;
+}
+
+
+
+
+void SubWindow::setBorderColor(const QColor &c)
+{
+	m_borderColor = c;
+}
+
+
+
+
 void SubWindow::moveEvent(QMoveEvent * event)
 {
 	QMdiSubWindow::moveEvent(event);
 	// if the window was moved and ISN'T minimized/maximized/fullscreen,
 	//   then save the current position
-	if (!isMaximized() && !isMinimized() && !isFullScreen())
+	if(!isMaximized() && !isMinimized() && !isFullScreen())
 	{
 		m_trackedNormalGeom.moveTopLeft(event->pos());
 	}
 }
 
+
+
+
 void SubWindow::resizeEvent(QResizeEvent * event)
 {
+	/* button adjustments*/
+	m_minimizeBtn->hide();
+	m_maximizeBtn->hide();
+	m_restoreBtn->hide();
+
+	const int rightSpace = 3;
+	const int buttonGap = 1;
+	const int menuButtonSpace = 24;
+
+	QPoint rightButtonPos( width() - rightSpace - m_buttonSize.width() , 3 );
+	QPoint middleButtonPos( width() - rightSpace - ( 2 * m_buttonSize.width() ) - buttonGap, 3 );
+	QPoint leftButtonPos( width() - rightSpace - ( 3 * m_buttonSize.width() ) - ( 2 * buttonGap ), 3 );
+
+	//The buttonBarWidth relates on the count of button.
+	//We need it to calculate the width of window title label
+	int buttonBarWidth = rightSpace + m_buttonSize.width();
+
+	//set the buttons on their positions.
+	//the close button is ever needed and on the rightButtonPos
+	m_closeBtn->move(rightButtonPos);
+
+	//here we ask: is the Subwindow maximizable and/or minimizable
+	//then we set the buttons and show them if needed
+	if ( windowFlags() & Qt::WindowMaximizeButtonHint )
+	{
+		buttonBarWidth = buttonBarWidth + m_buttonSize.width() + buttonGap;
+		m_maximizeBtn->move(middleButtonPos);
+		m_maximizeBtn->show();
+	}
+	if ( windowFlags() & Qt::WindowMinimizeButtonHint )
+	{
+		buttonBarWidth = buttonBarWidth + m_buttonSize.width() + buttonGap;
+		if (m_maximizeBtn->isHidden())
+		{
+			m_minimizeBtn->move(middleButtonPos);
+		}
+		else
+		{
+			m_minimizeBtn->move(leftButtonPos);
+		}
+		m_minimizeBtn->show();
+		m_restoreBtn->hide();
+		if ( isMinimized() )
+		{
+			if (m_maximizeBtn->isHidden())
+			{
+				m_restoreBtn->move(middleButtonPos);
+			}
+			else
+			{
+				m_restoreBtn->move(leftButtonPos);
+			}
+			m_restoreBtn->show();
+			m_minimizeBtn->hide();
+		}
+	}
+
+	// title QLabel adjustments
+	m_windowTitle->setAlignment( Qt::AlignHCenter );
+	m_windowTitle->setFixedWidth( widget()->width() - (menuButtonSpace + buttonBarWidth) );
```

Comment:
Same applies to space after/before parentheses. Either is fine, but you have to pick one.